data/network/src/commonMain/kotlin/social/androiddev/common/network/MastodonApiImpl.kt

```diff
@@ -13,12 +13,15 @@ import io.ktor.client.HttpClient
 import io.ktor.client.call.body
 import io.ktor.client.plugins.ResponseException
 import io.ktor.client.request.get
+import io.ktor.client.request.post
+import io.ktor.client.request.forms.formData
 import kotlinx.serialization.SerializationException
+import social.androiddev.common.network.model.Application
 import social.androiddev.common.network.model.Instance
+import social.androiddev.common.network.util.runCatchingIgnoreCancelled
 
 class MastodonApiImpl(
```

Comment:
For discussion: https://octoperf.com/blog/2016/10/27/impl-classes-are-evil/#:~:text=Impl%20is%20meaningless&text=If%20you%20can%20find%20a,breaking%20the%20Single%20Responsibility%20principle.

In general, a public class called Impl is a bit of a smell to me. Are we intending to have multiple instances of this? Or is it solely here to allow for easier testing? If it is the latter, I'd like to recommend using a static constructor on `MastodonApi` and make this class `internal`

Reply:
I agree 100%
Afaik very less people understand this principle, that's why I started with the mostly known `*Impl` naming to have it as explicit and easy to understand for everyone. But since we finally decided for a DI framework and have a good start of a documentation, we can change it and redirect people to our documentation.

Reply:
+1 - I was typing up a response to say similar. And for now I just copied this same pattern for the repository until we got DI setup etc. I didn't want to add even more changes to this PR beyond what I did. A good follow-up should be the DI framework though now that we agreed on the library. I can do that and then sweep through and fix both this and the Repository impl